apache2/waf_logging/waf_log_util.cc

```diff
@@ -91,77 +52,59 @@ void set_waf_format(waf_format::Waf_Format* waf_format, char* resourceId, char*
 				}
 			}
 			mandatory_message[ind + i] = '\0';
-			properties->set_message(mandatory_message);
+			message_str = mandatory_message;
 		}
 		else {
 			messages[strlen(messages) - 1] = '\0';
-			properties->set_message(messages+1);
+			message_str = messages+1;
 		}		
 	}
 
 	if (site == 0) {
-        properties->set_site(waf_format::Properties::Global);
+        site_str = "Global";
     }
-    
-    if (details_messages != NULL) {
-        details->set_message(details_messages);
+
+    if (details_file != NULL) {
+		details_file[strlen(details_file) - 1] = '\0';
     }
-    
+
     if (details_data != NULL) {
 		details_data[strlen(details_data) - 1] = '\0';
-        details->set_data(details_data+1);
-    }
-    
-    if (details_file != NULL) {
-        details->set_file(details_file);
     }
     
     if (details_line != NULL) {
 		details_line[strlen(details_line) - 1] = '\0';
-        details->set_line(details_line+1);
     }
-    
-	if (hostname != NULL) {
-		properties->set_hostname(hostname);
-	}
 
 	if (waf_unique_id != NULL) {
 		waf_unique_id[strlen(waf_unique_id) - 1] = '\0';
-		properties->set_transactionid(waf_unique_id+1);
 	}
+
+    waf_logging::waf_log waf_log{resourceId, operationName, category, waf_logging::property{ instanceId, clientIP, clientPort, requestUri, ruleSetType, ruleSetVersion, ruleId+1, message_str, action_str, site_str, waf_logging::detail{ details_messages, details_data+1, details_file+1, details_line+1 }, hostname, waf_unique_id+1}};
```

Comment:
(minor) You can just write 
```
return {resourceId, operationName, ...., waf_unique_id+1}};
```
instead of this and the next line. Would be a bit shorter.

Reply:
will change